docker_autoscaler.tf

```diff
@@ -0,0 +1,178 @@
+#
+# This file is responsible for creating the resources needed to run the docker autoscaler plugin from GitLab. It replaces the
+# outdated docker+machine driver. The docker+machine driver is a legacy driver that is no longer maintained by GitLab.
```

Comment:
Good addition.
"a legacy driver that is no longer maintained by GitLab."
-> "a legacy driver that is no longer maintained by Docker."
OU
-> "a legacy driver that will soon no longer be maintained by Gitlab."

From Gitlab official documentation:

"This is a fork of Docker Machine maintained by GitLab for fixing critical bugs. Docker Machine, which Docker has deprecated as of 2021-09-27, is the basis of the [GitLab Runner Docker Machine Executor](https://docs.gitlab.com/runner/configuration/autoscale.html).
We will maintain the fork and provide support through the end of the calendar year 2024."